Hub/Services/PlanNode.cs

```diff
@@ -339,9 +340,12 @@ public IEnumerable<ActivityTemplateDTO> GetSolutions(IUnitOfWork uow)
 
         public IEnumerable<ActivityTemplateCategoryDTO> GetAvailableActivityGroups()
         {
+            var availableTerminalIds = _terminal.GetAll().Select(x=>x.Id).ToList();
+
             var curActivityTemplates = _activityTemplate
                 .GetQuery()
-                .Where(at => at.ActivityTemplateState == ActivityTemplateState.Active).AsEnumerable().ToArray()
+                .Where(at => at.ActivityTemplateState == ActivityTemplateState.Active 
```

Comment:
ActivityTemplates filtering is better to be done in the ActivityTemplate service. This will make security check more reliable and predictable